pages/dkp/kommander/2.1/projects/applications/catalog-applications/custom-resources-workspace-catalog/spark-operator-custom-resources/index.md

```diff
@@ -0,0 +1,135 @@
+---
+layout: layout.pug
+navigationTitle: Deploy Custom Resources with Spark Operator
+title: Deploy Custom Resources with Spark Operator
+menuWeight: 20
+excerpt: Get Started on Spark Operator Custom Resources
+---
+
+To run your Spark workloads with Spark Operator, apply the Spark Operator specific custom resources. The Spark Operator works with the following  kinds of custom resources:
+
+- `SparkApplication`
+- `ScheduledSparkApplication`
+
+See [Spark Operator API documentation](https://github.com/mesosphere/spark-on-k8s-operator/blob/d2iq-master/docs/api-docs.md) for more details.
+
+## Prerequisites
+
+1.  First, deploy your Spark Operator. See the [Spark Operator](../../../../../workspaces/applications/catalog-applications/spark-operator) documentation for more information.
+
+2.  You need to ensure the necessary RBAC resources referenced in your custom resources exist, otherwise the your custom resources will fail. See [this Spark Operator documentation](https://github.com/mesosphere/spark-on-k8s-operator/blob/d2iq-master/docs/quick-start-guide.md#about-the-spark-job-namespace) for details.
+    -   This is an example of commands you need to create RBAC resources needed in your project namespace:
+
+        ```bash
+        export PROJECT_NAMESPACE=<project namespace>
+        kubectl apply -f - <<EOF
+        apiVersion: v1
+        kind: ServiceAccount
+        metadata:
+        name: spark-service-account
+        namespace: ${PROJECT_NAMESPACE}
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: Role
+        metadata:
+        namespace: ${PROJECT_NAMESPACE}
+        name: spark-role
+        rules:
+        - apiGroups: [""]
+        resources: ["pods"]
+        verbs: ["*"]
+        - apiGroups: [""]
+        resources: ["services"]
+        verbs: ["*"]
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: RoleBinding
+        metadata:
+        name: spark-role-binding
+        namespace: ${PROJECT_NAMESPACE}
+        subjects:
+        - kind: ServiceAccount
+        name: spark-service-account
+        namespace: ${PROJECT_NAMESPACE}
+        roleRef:
+        kind: Role
+        name: spark-role
+        apiGroup: rbac.authorization.k8s.io
```

Comment:
I think this yaml is not formatted correctly - an editor might have cobbled up the indentations. 

This is my first wack at it, but it probably exists somewhere that might be easier to copy/paste.

```suggestion
        apiVersion: v1
        kind: ServiceAccount
        metadata:
          name: spark-service-account
          namespace: ${PROJECT_NAMESPACE}
        ---
        apiVersion: rbac.authorization.k8s.io/v1
        kind: Role
        metadata:
          namespace: ${PROJECT_NAMESPACE}
          name: spark-role
          rules:
          - apiGroups: [""]
            resources: ["pods"]
            verbs: ["*"]
          - apiGroups: [""]
            resources: ["services"]
            verbs: ["*"]
        ---
        apiVersion: rbac.authorization.k8s.io/v1
        kind: RoleBinding
        metadata:
          name: spark-role-binding
          namespace: ${PROJECT_NAMESPACE}
          subjects:
          - kind: ServiceAccount
            name: spark-service-account
            namespace: ${PROJECT_NAMESPACE}
          roleRef:
          kind: Role
            name: spark-role
            apiGroup: rbac.authorization.k8s.io
```